src/components/Truncate/Truncate.tsx

```diff
@@ -76,18 +76,30 @@ export class Truncate extends React.PureComponent<
       this.contentNode.style.display = 'initial'
       // 2. Calculate the differences
       const isContentTruncated =
-        // TODO: fix typescript complains
-        // @ts-ignore
-        this.contentNode.offsetWidth > this.node.offsetWidth
+        props.splitter && props.showTooltipOnTruncate
+          ? this.isSplitContentTruncated(this.contentNode, this.node)
+          : // TODO: fix typescript complains
+            // @ts-ignore
+            this.contentNode.offsetWidth > this.node.offsetWidth
       // 3. Resets the display
       // TODO: fix typescript complains
       // @ts-ignore
-      this.contentNode.style.display = null
 
       return isContentTruncated
     }
   }
 
+  isSplitContentTruncated = (contentNode: any, node: any): boolean => {
```

Comment:
I'm not sure what to put here so TS doesn't complain. Property `offsetWidth` doesn't exist on type Node or Element. I also get the error `HTML collection is not an array type` for type Element.

Reply:
Have you tried `HTMLElement`? I wouldn't worry too much about it, sometimes TS just likes to be troublesome and not worth wasting your time.

Reply:
With the `HTMLElement` type it complains that "Object is possibly null"  for `this.node` and "Argument of type 'null' is not assignable to parameter of type 'HTMLElement'" for `this.contentNode`. 🤔

Reply:
It looks like this is the correct behavior of TypeScript because it can be null. In fact, we actually initialize those variables with null. I found this article on the topic really useful: https://medium.com/@martin_hotell/react-refs-with-typescript-a32d56c4d315 - TLDR: it seems we can satisfy the TypeScript compiler by guarding the use of the variables with if conditions.